### PR TITLE
[DOCS] Add clear scroll API reference docs

### LINKED ISSUE #56833: [DOCS] Scroll APIs are missing from REST API Docs
The scroll APIs are currently documented in the request body search docs, which are marked for removal.

The scroll tutorials will be migrated as part of #48194. However, we should also add reference-style documentation for these APIs:

- [x] Scroll API #57153
- [ ] Clear scroll API #57367

### PR DESCRIPTION
**Pull request by jrodewig**
Adds reference documentation for the clear scroll API.

This supplements existing docs in the [Scroll][0] section.

Closes #56833

[0]: https://www.elastic.co/guide/en/elasticsearch/reference/master/search-request-body.html#request-body-search-scroll

### CODE REVIEW

File: docs/reference/search/clear-scroll-api.asciidoc [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+`DELETE /_search/scroll/<scroll_id>`
+deprecated:[7.0.0]
+
+`DELETE /_search/scroll`
+
+[[clear-scroll-api-path-params]]
+==== {api-path-parms-title}
+
+`<scroll_id>`::
+deprecated:[7.0.0]
```
Comment (mayya-sharipova) [inline docs/reference/search/clear-scroll-api.asciidoc:46]: @jrodewig  Can you please share how did you know that <scroll_id> is deprecated as a path param?

Comment (jrodewig) [reply docs/reference/search/clear-scroll-api.asciidoc:46]: Hi @mayya-sharipova. While the API doesn't return a deprecation warning, the JSON spec indicates that this path param is deprecated:
https://github.com/elastic/elasticsearch/blame/master/rest-api-spec/src/main/resources/rest-api-spec/api/clear_scroll.json#L28

If that's not the case, I can update these docs and fix the JSON spec too. Thanks!

Comment (mayya-sharipova) [reply docs/reference/search/clear-scroll-api.asciidoc:46]: Thanks @jrodewig. I have confirmed with es clients team that we indeed deprecating scroll_id as a path param in DELETE API.

**Review by mayya-sharipova — APPROVED** [review]
@jrodewig thanks.

**Comment by jrodewig** [comment]
Thanks @mayya-sharipova

**Comment by jrodewig** [comment]
### Backport commits
master 69b79d2
7.x 80faafd 
7.8 0684f9c 
7.7 1998fe5

**Pull request merged**